Solutions/Branwen/GUI.cs

```diff
@@ -0,0 +1,166 @@
+                    sheets.Append(sheet);
+                    WriteDirectoryToWorksheet(RunInventory(topLevelDirectories[i]), sheetData);
+                }
+                spreadsheetDocument.Close();
+                MessageBox.Show("DONE! Files Inventoried:  " + fileCount);
+                buttonSelectAndRunInventory.Enabled = true;
+                buttonSelectAndRunInventory.Text = "Select Inventory Directory";
+            }
+            catch (Exception ex)
+            {
+                MessageBox.Show("Failed to Inventory:" + Environment.NewLine + ex.Message);
+                buttonSelectAndRunInventory.Enabled = true;
+                buttonSelectAndRunInventory.Text = "Select Inventory Directory";
+            }
+        }
+
+        /// <summary>
+        /// Recursively finds all Files in the parent directory
+        /// </summary>
+        /// <param name="parent"></param>
+        /// <returns></returns>
+        private List<FileInfo> RunInventory(DirectoryInfo parent)
+        {
+            List<FileInfo> toReturn = new List<FileInfo>();
+            foreach (DirectoryInfo directory in parent.GetDirectories())
+            {
+                toReturn.AddRange(RunInventory(directory));
+            }
+            toReturn.AddRange(parent.GetFiles());
+            return toReturn;
+        }
+
+        /// <summary>
+        /// Writes all files in the Directory to the Workbook
+        /// </summary>
+        /// <param name="files"></param>
+        /// <param name="writer"></param>
+        private void WriteDirectoryToWorksheet(List<FileInfo> files, SheetData sheetData)
+        {
+            //Write Worksheet Header
+            WriteHeader(sheetData);
+
+            for (int i = 0; i < files.Count; i++)
+            {
+                Row row = new Row();
+                Cell cell = new Cell();
+                
+                //File Name
+                cell.CellValue = new CellValue(Path.GetFileNameWithoutExtension(files[i].Name));
+                cell.DataType = CellValues.String;
+                row.Append(cell);
+
+                //File Extension
+                cell = new Cell();
+                cell.CellValue = new CellValue(Path.GetExtension(files[i].Name));
+                cell.DataType = CellValues.String;
+                row.Append(cell);
+
+                //Filesize
+                cell = new Cell();
+                cell.CellValue = new CellValue((files[i].Length / 1048576).ToString());
+                cell.DataType = CellValues.Number;
+                row.Append(cell);
+
+                //write all the pathNames to the spreadsheet
+                List<string> path = files[i].DirectoryName.Split('\\').ToList(); ;
+                foreach (string pathName in path)
+                {
+                    cell = new Cell();
+                    cell.CellValue = new CellValue(pathName);
+                    cell.DataType = CellValues.String;

```

Comment:
you are setting the value to the same thing in every iteration, just remove it from the loop